### PR TITLE
prefer more generic Mapping over dict type

### PR DESCRIPTION
**Pull request by edvardm**
typically it's better to specify more generic types for input parameters, and specific types our return values. In this case changing headers param from dict to Mapping should resolve typing issues when eg. starlette.requests.headers.Headers instance is passed to get_principal_from_headers()

### CODE REVIEW

File: audit_log/headers.py [hunk 1]
```diff
@@ -65,14 +66,14 @@ def parse_spiffe(xfcc_header: str) -> ParsedSPIFFE:
...
-    headers: dict[str, str],
+    headers: typing.Mapping[str, str],
 ) -> Principal:
     """Get principal from headers, supports mTLS, headers set in Istio, and JWTs.
 
     Note: Do not use this to handle your auth, it expects auth to already be handled elsewhere and this is just to help get principals.
 
     Args:
-        headers (dict[str, str]): Headers with all keys lowercase
+        headers (Mapping[str, str]): Headers with all keys lowercase
```
Comment (edvardm) [inline audit_log/headers.py:76]: as we normalize headers by lowercasing those, we could remove "with all keys lowercase" requirement from doc too? Could be separate PR though

File: audit_log/headers.py [hunk 2]
```diff
@@ -65,14 +66,14 @@ def parse_spiffe(xfcc_header: str) -> ParsedSPIFFE:
 
 
 def get_principal_from_headers(
-    headers: dict[str, str],
+    headers: typing.Mapping[str, str],
```
Comment (jerr0328) [inline audit_log/headers.py:69]: typing.Mapping is deprecated since 3.9: https://docs.python.org/3/library/typing.html#typing.Mapping

File: audit_log/headers.py (hunk 1 above)
Comment (jerr0328) [reply audit_log/headers.py:76]: Yes that should be changed, the function will recreate a dict with lowercase keys

Comment (jerr0328) [reply audit_log/headers.py:76]: Something like this maybe for the comment?
```suggestion
        headers (Mapping[str, str]): HTTP Headers mapping (note that function will convert all keys to lowercase so case doesn't matter)
```
